### DISCUSSION TITLE
Beets via nix on M1 mac

### DISCUSSION THREAD

**Discussion by jmmaloney4** [beetbox/beets#4442] [General]
When trying to install beets on my M1 mac via nix I get the following error:
```
error: Package ‘beets-1.6.0’ in /nix/store/b3qhx504lxsmfagqla0jfn8mj866px2r-nixpkgs/nixpkgs/pkgs/tools/audio/beets/common.nix:170 is not supported on ‘aarch64-darwin’, refusing to evaluate.
```

Is there a reason beets is not supported on M1 macs, or is this just a configuration issue?

Thanks,
Jack

**Comment by wisp3rwind** [comment]
I'm not aware of any reason directly within beets that should prevent running on ARM. There could of course be a non-portable dependency, I've never checked that. Maybe, it'd be easiest to check with the nix packagers whether there's an explicit reason not to support the architecture, or whether just nobody enabled it?

**Comment by jackwilsdon** [comment]
It seems like the package is set to support Linux only:

https://github.com/NixOS/nixpkgs/blob/bd0a5ce380fee429032a70b75160b1261d38c868/pkgs/tools/audio/beets/common.nix#L174

It may be worth trying to change this locally and see if it runs alright - it could just be that it has never been tested on macOS before.

**Comment by tasmo** [comment]
With `pyenv` installed Python and `pip` installation of `beets` it works well on Darwin ARM64.
